src/opentime/timeRange.h

```diff
@@ -45,57 +54,199 @@ class TimeRange {
     }
 
     TimeRange duration_extended_by(RationalTime other) const {
-        return TimeRange {_start_time, _duration + other};
+        return TimeRange{_start_time, _duration + other};
     }
 
     TimeRange extended_by(TimeRange other) const {
-        RationalTime new_start_time {std::min(_start_time, other._start_time)},
-                     new_end_time {std::max(end_time_exclusive(), other.end_time_exclusive())};
+        RationalTime new_start_time{std::min(_start_time, other._start_time)},
+                new_end_time{std::max(end_time_exclusive(), other.end_time_exclusive())};
 
-        return TimeRange {new_start_time,
-                          RationalTime::duration_from_start_end_time(new_start_time, new_end_time)};
+        return TimeRange{new_start_time,
+                         RationalTime::duration_from_start_end_time(new_start_time, new_end_time)};
     }
 
     RationalTime clamped(RationalTime other) const {
         return std::min(std::max(other, _start_time), end_time_inclusive());
     }
 
     TimeRange clamped(TimeRange other) const {
-        TimeRange r {std::max(other._start_time, _start_time), other._duration};
-        RationalTime end {std::min(r.end_time_exclusive(), end_time_exclusive())};
-        return TimeRange {r._start_time, end - r._start_time};
+        TimeRange r{std::max(other._start_time, _start_time), other._duration};
+        RationalTime end{std::min(r.end_time_exclusive(), end_time_exclusive())};
+        return TimeRange{r._start_time, end - r._start_time};
     }
 
+    /**
+     * These relations implement James F. Allen's thirteen basic time interval relations.
+     * Detailed background can be found here: https://www.ics.uci.edu/~alspaugh/cls/shr/allen.html
+     */
+
+    /**
+     * In the relations that follow, epsilon indicates the tolerance,in the sense that if abs(a-b) < epsilon,
+     * we consider a and b to be equal
+     */
+
+    /**contains(other)
+     * params: RationalTime other
+     * The start of self precedes other.
+     * other precedes the end of self.
+     */
     bool contains(RationalTime other) const {
         return _start_time <= other && other < end_time_exclusive();
     }
 
+    /**contains(other)
+     * params: TimeRange other
+     * The start of self precedes start of other.
+     * The end of self antecedes end of other.
+     * The converse would be other.contains(self)
+     */
```

Comment:
I think this style is fine, but we should move to a doxygen style comment.  I'll file an issue for that and describe that as a project there.